oci/layout/oci_src.go

```diff
@@ -132,6 +137,16 @@ func (s *ociImageSource) GetBlob(ctx context.Context, info types.BlobInfo, cache
 	return r, fi.Size(), nil
 }
 
+func containsValidHTTPURLs(urls []string) bool {
+	for _, s := range urls {
+		u, err := url.Parse(s)
+		if err == nil && (u.Scheme == "http" || u.Scheme == "https") {
```

Comment:
I'd prefer to have `getExternalBlob` only attempt to fetch the URL if it's a supported scheme.  Could we instead extend `getExternalBlob` to do that?  Same for the other case.

Reply:
Thank you for the review. Fixed `getExternalBlob`.